.github/workflows/release.yml

```
@@ -37,7 +42,7 @@ jobs:
         push: true
         tags: rancher/rke2-upgrade:${{ env.GIT_TAG }}
         file: Dockerfile
-        platforms: linux/amd64
+        platforms: ${{ matrix.os }}/${{ matrix.arch }}
```

Don't do this. If you do the matrix in the job, it will run the publish action separately for each combination in the matrix, with the result from each run overwriting the results from the previous one.

You should instead run it once, but pass `platforms` a list of the images you want to build. You will probably need to modify the Dockerfile to use TARGETARCH instead of ARCH, as this is what is passed in to the docker build step for each arch in the list.
```suggestion
        platforms: linux/amd64, linux/arm64
```